yamllint/rules/comments.py

```diff
@@ -97,7 +97,9 @@ def check(conf, comment):
                     comment.column_no == 1 and
                     re.match(r'^!\S', comment.buffer[text_start:])):
                 return
-            elif comment.buffer[text_start] not in (' ', '\n', '\0'):
+            # We can test for both \r and \r\n just by checking first char
+            # \r itself is a valid newline on some older OS.
```

Comment:
OK, the problem becomes simpler then.

This does no handle examples like:

    ---\n
    #\rcomment\n

and:

    ---\r\n
    #\r[end of file]

... but I guess it's acceptable to ignore those.

Reply:
> but I guess it's acceptable to ignore those.

Agreed. I don't think mixing of newline types in those corner cases would be something you'd want yamllint to be okay with.